### DISCUSSION TITLE
Limit user permissions to editing items from defined list of other users

### DISCUSSION THREAD

**Discussion by luciandavidescu** [directus/directus#17491] [Feature Requests]
### Summary

A user's capabilities can be limited to see/update/delete only his own set of items, with a filter rule such as _User Created » Equals » `$CURRENT_USER`_.

Next level is a user who can see/update/delete items from a specific role or from any other user.

How about an intermediate level, where a user is allowed to see/update/delete the items of a certain group of certain other users but not all.

### Basic Example

Easiest and most intuitive way to do it would likely be to have a _many to many_ field where the other users would be added (e.g. 'team') and that could be accessed with a filter like _User Created » Is one of » `$CURRENT_USER.team`_

### Motivation

In production environments with medium-to-large teams, this would reflect the fact that mid-level managers have editing authority only for their own subordinates, not for everybody who shares the same rank.

### Detailed Design

The `$CURRENT_USER` parameter should make the actual fields of the current user available, so they can be retrieved with `$CURRENT_USER.field_name`

### Must Have:
Ability to access an item's subfields from permissions

### Should Have:
-

### Could Have:
-

### Won't Have:
-

### Drawbacks

Complexity unknown

### Alternatives

Creating a separate role for each team. This would have much less flexibility and would amount to effectively repurposing the feature.

### Adoption Strategy

Should have no impact

**Comment by rijkvanzanten** [comment]
> Easiest and most intuitive way to do it would likely be to have a many to many field where the other users would be added (e.g. 'team') and that could be accessed with a filter like User Created » Is one of » $CURRENT_USER.team

This should already be possible! `$CURRENT_USER` and `$CURRENT_ROLE` accept dot-notation to access specific fields, so if you were to add a custom field to Directus Users, you can access it exactly like you're proposing in permissions 🙂

**Reply by luciandavidescu** [reply]
Must be missing something badly because surely I tried that... Does this filter make sense? _User Created » Is one of » $CURRENT_USER.team_ (where 'team' is a many to many field between directus_users and itself)

**Reply by rijkvanzanten** [reply]
`team` in this case would be an array of junction rows on the many to many table, so that wouldn't result in what you'd expect. Can you try `$CURRENT_USER.team.user_id` (assuming your junction table has a column `user_id` that holds the user's UUID)?

**Reply by luciandavidescu** [reply]
Thanks, makes sense now. So I looked into the DB and the junction table Directus generated has 3 columns – id (integer) directus_users_id (current user UUID) and related_directus_users_id (team UUIDs). So I did `User Created » Is one of » $CURRENT_USER.team.related_directus_users_id ` and surely it works.
